packages/ember-data/lib/system/model/model.js

```diff
@@ -1091,64 +691,14 @@ var Model = Ember.Object.extend(Ember.Evented, {
     with an error.
   */
   reload: function() {
-    set(this, 'isReloading', true);
-
-    var record = this;
-    var promiseLabel = "DS: Model#reload of " + this;
-    var promise = new Promise(function(resolve) {
-      record.send('reloadRecord', resolve);
-    }, promiseLabel).then(function() {
-      record.set('isError', false);
-      return record;
-    }, function(reason) {
-      record.set('isError', true);
-      throw reason;
-    }, "DS: Model#reload complete, update flags")['finally'](function () {
-      record.set('isReloading', false);
-      record.updateRecordArrays();
-    });
-
+    var model = this;
     return PromiseObject.create({
-      promise: promise
+      promise: this._internalModel.reload().then(function() {
```

Comment:
what is reloads normal fulfillment value, seems like `model` would be a good default ?